### DISCUSSION TITLE
[ "Nice to have... " ]  Pico-W and Pico2-W support?

### DISCUSSION THREAD

**Discussion by DeeJay** [lurk101/pshell#29] [Ideas]
_[Edit: since typing the following less than an hour ago I have heard rumours via PiMoRoNi of an RPi RM2 wireless module compatible with rp2350. So the target implementation may be about to change... ]_

Not in any way essential or detracting from the value of the work you have done.

Might you consider - or provide a hint as to how to self-provide - a build of pshell that would run on the **W** variants of the pico boards to the extent that the on-board LED can be controlled? (Not expecting BT or wlan support.)

The released code as of today seems to run perfectly well on an rp2040 Pico-W, but without LED support.

I understand very little of the RPi Pico SDK. I have tried building with
`cmake .. -DPICO_BOARD=pico_w -DPICO_PLATFORM=rp2040
`
 
but that fails with

```
In file included from /home/david/Projects/PicoW/pshell/cc/cc.c:257:
/home/david/Projects/PicoW/pshell/cc/2040/cc_defs.h:51:30: error: 'PICO_DEFAULT_LED_PIN' undeclared here (not in a function); did you mean 'PICO_DEFAULT_SPI'?
   51 |     {"PICO_DEFAULT_LED_PIN", PICO_DEFAULT_LED_PIN},
```

I have tried modifying cc_defs.h to use the CYW43 version of the function, but not to a point where the project will build and run. Possibly it is finding the headers but not  the actual binary of the supporting library?

Thanks for listening...

(My selfish motivation is that since the price delta for the W variants is so slim there seems to be no reason to buy the base networkless version for home hacking any more.)

**Comment by lurk101** [comment]
Unfortunately I do not have a PicoW to work with.

**Reply by DeeJay** [reply]
Have noticed this promising development.

> /: news
> 
> What's new in version 2.1.5
> 
> General
> 
>  - general source code, cmake, and script file cleanup
>  - **add support for picow** and vgaboard
>  - update C example source code
> 


Source can be built for, and runs on, RPi Pico_W

Thank you for this.


But cc-examples/blink.c does not work.

```
/: cc blink.c

Error : undefined variable PICO_DEFAULT_LED_PIN
4:     int led_pin = PICO_DEFAULT_LED_PIN;

/: 

```

Looking into whether the cyw43 definition of the LED is supported.


It isn't straightforward because the led is driven by the wireless chip, and needs calls to driver functions in #include "pico/cyw43_arch.h"

Need cyw43_arch_init() and cyw43_arch_gpio_put()



Would it help to procure a pico_w for you, and maybe a pico2_w when available, or to make a donation?
